src/frontend/screens/Settings/MapSettings/MapImportBottomSheet.tsx

```diff
@@ -0,0 +1,326 @@
+    id: "screens.Settings.MapSettings.fileErrorDescription",
+    defaultMessage: "Error importing file, please try a different file.",
+    description: "Description for file error in bottom sheet content",
+  },
+  defaultMap: {
+    id: "screens.Settings.MapSettings.defaultMap",
+    defaultMessage: "Default Map",
+    description: "Name of default map",
+  },
+});
+
+/**
+ * Bottom sheet content for the user to select a file to import
+ */
+const SelectImportContent = ({
+  onPressImport,
+  onPressClose,
+}: {
+  onPressImport: () => void;
+  onPressClose: () => void;
+}) => {
+  const { formatMessage: t } = useIntl();
+  return (
+    <BottomSheetContent
+      title={t(m.BackgroundMapTitle)}
+      buttonConfigs={[
+        {
+          onPress: onPressClose,
+          text: t(m.close),
+          variation: "outlined",
+        },
+      ]}
+    >
+      <TouchableOpacity onPress={onPressImport} style={styles.importButton}>
+        <View style={styles.importTextAndIcon}>
+          <MaterialIcon name="file-upload" size={30} color={MEDIUM_GREY} />
+          <Text style={styles.text}> {t(m.importFromFile)}</Text>
+        </View>
+        <Text style={[styles.text, { textAlign: "center" }]}>
+          {"( .mbtiles )"}
+        </Text>
+      </TouchableOpacity>
+    </BottomSheetContent>
+  );
+};
+
+/**
+ * Bottom sheet content for when an imported file is processing, checking the
+ * format etc. is correct (shows until the import endpoint returns with the
+ * import id). User is unable to close this sheet.
+ */
+const ImportingContent = () => {
+  return <BottomSheetContent title="Processing file..." buttonConfigs={[]} />;
+};
+
+/**
+ * Bottom sheet content for when there is an error with the selected file
+ */
+const FileErrorContent = ({ onPressClose }: { onPressClose: () => void }) => {
+  const { formatMessage: t } = useIntl();
+  return (
+    <BottomSheetContent
+      title={t(m.importErrorTitle)}
+      icon={<ErrorIcon color={RED} size={90} style={styles.errorIcon} />}
+      description={t(m.fileErrorDescription)}
+      buttonConfigs={[
+        {
+          onPress: onPressClose,
+          text: t(m.close),
+          variation: "outlined",
+        },
+      ]}
+    />
+  );
+};
+
+type BottomSheetStates = "select" | "importing" | "error";
+
+export interface MapImportBottomSheetMethods {
+  /** Open the bottom sheet by expanding it to its maximum snap point */
+  open: () => void;
+}
+
+/** Custom backdrop that will close sheet when pressed */
+const BackdropPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"close"} />
+  );
+};
+
+/** Custom backdrop that will not close sheet when pressed */
+const BackdropNonPressable = (props: BottomSheetBackdropProps) => {
+  return (
+    <BottomSheetBackdrop {...props} opacity={0.3} pressBehavior={"none"} />
+  );
+};
+
+const MapImportBottomSheet = React.forwardRef<MapImportBottomSheetMethods, {}>(
```

Comment:
depending on the behavior we want, this could reuse our BottomSheetModal component. using the BottomSheet was to account for issues with collapsing/closing and re-opening the sheet in sequence, which was only necessary because of my initial implementation of the error-handling.

think with Gregor's refactor and the UX changes for errors on this screen, we could scrap the usage of the BottomSheet and stick with BottomSheetModal, since that more closely reflects the usage we want (sheet that has blocking actions)